app.ts

```diff
@@ -0,0 +1,388 @@
+$(".search-input").on("input", function (this: HTMLInputElement, event: any) {
+	if (isFunctionRunning) {
+		return;
+	}
+	isFunctionRunning = true;
+	event.preventDefault();
+	return apiManager.getRecordCount()
+		.then((count) => {
+			let inputNumber = input();
+			let inputNumberInt: any = parseInt(inputNumber);
+			if (inputNumber !== '') {
+				const maxRecords = recordsPerPage();
+				let end = Math.ceil(inputNumberInt / maxRecords) * maxRecords;
+				if (end > (count + 1)) {
+					end = count;
+					globals.currentToID = end;
+				}
+				let start = (end - (maxRecords - 1));
+				globals.currentPage = Math.floor(end / maxRecords);
+				if (inputNumberInt < 1000000 && inputNumberInt > 0) {
+					if (end === 1000000) {
+						end = end - 1;
+					} else {
+						null;
+					}
+					$('.results-box').remove();
+					$('.search-container').append(`
+						<div class="results-box">
+							<p class="results-select">${start} - ${end}</p>
+						</div>`);
+					$('.results-box').on('click', resultsSelect);
+				} else {
+					$('.results-box').remove();
+					$('.search-container').append(`
+						<div class="results-box">
+							<p class="message">Invalid Input!</p>
+						</div>`);
+				}
+			} else {
+				$('.results-box').remove();
+			}
+			isFunctionRunning = false;
+		})
+		.catch((error) => {
+			throw error;
+		})
+})
+
+// After the range has been returned to the user. The user can click on it and that will show the range of records on the table. 
+function resultsSelect(event: any) {
+	if (isFunctionRunning) {
+		return;
+	}
+	isFunctionRunning = true;
+	let startID: number;
+	let endID: number;
+	let pageEnd: number;
+	let pageStart: number;
+	return apiManager.getRecordCount()
+		.then((count) => {
+			let idRange = $('.results-select').text();
+			let rangeArray = null;
+			rangeArray = idRange.split('-');
+			$('.results-box').remove();
+			startID = parseInt(rangeArray[0]);
+			endID = parseInt(rangeArray[1]);
+			let maxRecords = recordsPerPage();
+			globals.currentPage = Math.ceil(endID / maxRecords);
+			pageEnd = Math.ceil(globals.currentPage / 10) * 10;
+			pageStart = pageEnd - 9;
+			if (endID === count) {
+				startID = ((globals.currentPage - 1) * maxRecords) + 1;
+				endID = count;
+				globals.firstPage = pageStart;
+				globals.lastPage = pageEnd;
+			}
+		})
+		.then(() => {
+			return pageNumbers(pageStart, pageEnd);
+		})
+		.then(() => {
+			return showRecords(startID, endID);
+		})
+		.catch((error) => {
+			throw error;
+		})
+}
+
+// When adjusting the height and on different screen sizes. This function would responsible for calculating how much records should be displayed based on the height of the window itself. 
+function adjustDisplayedRecords() {
+	if (isFunctionRunning) {
+		return;
+	}
+	isFunctionRunning = true;
+	let screenHeight = $(window).height() as number;
```

Comment:
Please use `<>` to cast values. https://github.com/IMQS/www-lib/tree/master/_codestyle#type-casting